### PR TITLE
dragonball: Use vhost-net device by default

### PR DESCRIPTION
**Pull request by justxuewei**
This patch set vhost-net as default backend of networking. It allows users
to set `disable_vhost_net` to `true` to reenable virtio-net backend.
Plus, which backend to use is a matter of hypervisor, runtime-rs will no
longer need to know that.

Fixes: #8608

### CODE REVIEW

**Review by Apokleos — COMMENTED** [review]
Thx @justxuewei for your work!
But IMO you should split the PR into at least two commits, one for dragonball and one for runtime-rs, rather than having one PR which has only one commit with several changes at different levels.

File: src/dragonball/src/dbs_virtio_devices/src/vhost/vhost_kern/net.rs [hunk 1]
```diff
@@ -290,7 +290,9 @@ where
                 "{}: Invalid virtio queue pairs, expected a value greater than 0, but got {}",
                 NET_DRIVER_NAME, self.vq_pairs
             );
-            return Err(VirtioError::ActivateError(Box::new(ActivateError::InvalidParam)));
+            return Err(VirtioError::ActivateError(Box::new(
```
Comment (studychao) [inline src/dragonball/src/dbs_virtio_devices/src/vhost/vhost_kern/net.rs:293]: need a rebase here when the format commit from Dragonball is merged.

**Comment by justxuewei** [comment]
> Thx @justxuewei for your work! But IMO you should split the PR into at least two commits, one for dragonball and one for runtime-rs, rather than having one PR which has only one commit with several changes at different levels.

@Apokleos Thanks for your kindly review. I've split this patch into two commits, one for runtime-rs, another one for dragonball. PTAL.

**Review by studychao — APPROVED** [review]

File: src/runtime-rs/crates/hypervisor/src/dragonball/inner_device.rs [hunk 2]
```diff
@@ -209,9 +210,14 @@ impl DragonballInner {
         Ok(())
     }
 
-    fn add_net_device(&mut self, config: &NetworkConfig) -> Result<()> {
+    fn add_net_device(
+        &mut self,
+        hconfig: &HypervisorConfig,
```
Comment (Apokleos) [inline src/runtime-rs/crates/hypervisor/src/dragonball/inner_device.rs:215]: IMO, there's no need introduce a hconfig here, you can just get the hypervisor config in add_net_device before you do dragonball_network_device

Comment (Apokleos) [inline src/runtime-rs/crates/hypervisor/src/dragonball/inner_device.rs:215]: IMO, there's no need introduce a hconfig here, you can just get the hypervisor config in add_net_device before you do dragonball_network_device

File: src/runtime-rs/crates/hypervisor/src/dragonball/mod.rs [hunk 3]
```diff
@@ -201,41 +201,39 @@ impl Persist for Dragonball {
...
-            guest_mac: value.guest_mac.clone().map(|mac| {
-                // We are safety since mac address is checked by endpoints.
-                DragonballMacAddr::from_bytes(&mac.0).unwrap()
-            }),
-            use_shared_irq: value.use_shared_irq,
-            use_generic_irq: value.use_generic_irq,
-        }
+/// Generate Dragonball network config according to hypervisor config and
+/// runtime network config.
+pub(crate) fn dragonball_network_config(
```
Comment (Apokleos) [inline src/runtime-rs/crates/hypervisor/src/dragonball/mod.rs:206]: The function name ` dragonball_network_config` is not clear. I think it'd better to change it to something that is more specific and easier to understand, such as "build_xxx", "setup_xxx", "init_xxx"...

**Review by Apokleos — COMMENTED** [review]
Thx @justxuewei for your work! I have some comments.

File: src/runtime-rs/crates/hypervisor/src/dragonball/inner_device.rs (hunk 2 above)
Comment (justxuewei) [reply src/runtime-rs/crates/hypervisor/src/dragonball/inner_device.rs]: Fixed.

File: src/runtime-rs/crates/hypervisor/src/dragonball/mod.rs (hunk 3 above)
Comment (justxuewei) [reply src/runtime-rs/crates/hypervisor/src/dragonball/mod.rs]: Fixed.

**Review by Apokleos — APPROVED** [review]
Thx @justxuewei Nice work!
LGTM !

**Pull request merged**